src/altinn-app-frontend/src/components/base/ListComponent.tsx

```diff
@@ -26,110 +27,94 @@ const defaultDataList: any[] = [];
+  const dynamicDataList = useGetDataList({ id, mapping });
   const calculatedDataList = dynamicDataList || defaultDataList;
 
-  const rowsPerPage = useAppSelector(
-    (state) => state.dataListState.dataLists[dataListId || ''].size || pagination.default,
-  );
-  const currentPage = useAppSelector((state) => state.dataListState.dataLists[dataListId || ''].pageNumber || 0);
-  const sortColumn = useAppSelector((state) => state.dataListState.dataLists[dataListId || ''].sortColumn || null);
+  const rowsPerPage = useAppSelector((state) => state.dataListState.dataLists[id || ''].size || pagination.default);
+  const currentPage = useAppSelector((state) => state.dataListState.dataLists[id || ''].pageNumber || 0);
+  const sortColumn = useAppSelector((state) => state.dataListState.dataLists[id || ''].sortColumn || null);
   const sortDirection = useAppSelector(
-    (state) => state.dataListState.dataLists[dataListId || ''].sortDirection || SortDirection.NotActive,
+    (state) => state.dataListState.dataLists[id || ''].sortDirection || SortDirection.NotActive,
   );
   const totalItemsCount = useAppSelector(
-    (state) => state.dataListState.dataLists[dataListId || ''].paginationData?.totaltItemsCount || 0,
+    (state) => state.dataListState.dataLists[id || ''].paginationData?.totaltItemsCount || 0,
   );
-
-  const dataListHasChanged = useHasChangedIgnoreUndefined(dataList);
-  const { value, setValue } = useDelayedSavedState(handleDataChange, formData?.simpleBinding, 200);
-
-  React.useEffect(() => {
-    if (dataListHasChanged && formData.simpleBinding) {
-      // New datalist has been loaded, we have to reset form data.
-      // We also skip any required validations
-      setValue(undefined, true);
-    }
-  }, [dataListHasChanged, formData, setValue]);
-
-  const handleChange = ({ selectedValue }: altinnDesignSystem.ChangeProps) => {
+  const [value, setValue] = useState({});
+  const handleChange = ({ selectedValue }: ChangeProps) => {
     setValue(selectedValue);
+    for (const key in formData) {
+      handleDataChange(selectedValue[key], { key: key });
+    }
   };
 
-  const renderRow = (option) => {
+  const renderRow = (datalist) => {
     const cells: JSX.Element[] = [];
-    for (let i = 0; i < Object.keys(option).length; i++) {
-      cells.push(<TableCell key={i}>{option[Object.keys(option)[i]]}</TableCell>);
+    for (let i = 0; i < Object.keys(datalist).length; i++) {
+      cells.push(
+        <TableCell key={`${Object.keys(datalist)[i]}_${datalist[Object.keys(datalist)[i]]}`}>
+          {datalist[Object.keys(datalist)[i]]}
+        </TableCell>,
+      );
     }
     return cells;
   };
 
   const checkSortableColumns = (headers) => {
     const cell: JSX.Element[] = [];
-    let index = 0;
     for (const header of headers) {
       if ((sortableColumns || []).includes(header)) {
         cell.push(
           <TableCell
             onChange={handleSortChange}
             id={header}
```

Comment:
The id is just passed to TableCell in the designsystem. The id is supposed to be sat on the sorteble columns in case there is more than one sorteble column. Then only one of the columns is sorted a time.

Reply:
Ahh, right, I see that now. Maybe a bit unfortunate to mix those terms, as `id` is a well-recognized html attribute as well. I'd suggest renaming it to `sortKey` or something - but it's not really a big deal. :blush:

Reply:
I can change it :)